test/suite/test_layered32.py

```diff
@@ -0,0 +1,127 @@
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wttest
+from helper_disagg import DisaggConfigMixin, disagg_test_class, gen_disagg_storages
+from wtscenario import make_scenarios
+from wiredtiger import stat
+
+# test_layered32.py
+# Test that we write internal page deltas to the page log extension.
+
+@disagg_test_class
+class test_layered32(wttest.WiredTigerTestCase, DisaggConfigMixin):
+    encrypt = [
+        ('none', dict(encryptor='none', encrypt_args='')),
+        ('rotn', dict(encryptor='rotn', encrypt_args='keyid=13')),
+    ]
+
+    compress = [
+        ('none', dict(block_compress='none')),
+        ('snappy', dict(block_compress='snappy')),
+    ]
+
+    uris = [
+        ('layered', dict(uri='layered:test_layered32')),
+        ('btree', dict(uri='file:test_layered32')),
+    ]
+
+    ts = [
+        ('ts', dict(ts=True)),
+        ('non-ts', dict(ts=False)),
+    ]
+
+    conn_base_config = 'transaction_sync=(enabled,method=fsync),statistics=(all),statistics_log=(wait=1,json=true,on_close=true),' \
+                     + 'disaggregated=(page_log=palm),'
+    disagg_storages = gen_disagg_storages('test_layered32', disagg_only = True)
+
+    # Make scenarios for different cloud service providers
+    scenarios = make_scenarios(encrypt, compress, disagg_storages, uris, ts)
+
+    nitems = 1000
+
+    def session_create_config(self):
+        # The delta percentage of 200 is an arbitrary large value, intended to produce
+        # deltas a lot of the time.
+        cfg = 'disaggregated=(delta_pct=80),key_format=S,value_format=S,block_compressor={}'.format(self.block_compress)
+        if self.uri.startswith('file'):
+            cfg += ',block_manager=disagg'
+        return cfg
+
+    def conn_config(self):
+        enc_conf = 'encryption=(name={0},{1})'.format(self.encryptor, self.encrypt_args)
+        return self.conn_base_config + 'disaggregated=(role="leader"),' + enc_conf
+
+    # Load the storage store extension.
+    def conn_extensions(self, extlist):
+        extlist.extension('compressors', self.block_compress)
+        extlist.extension('encryptors', self.encryptor)
+        DisaggConfigMixin.conn_extensions(self, extlist)
+
+    def test_layered_read_write(self):
+        self.pr('CREATING')
```

Comment:
We should clean up these.

Reply:
@quchenhao Do you mean that we don't want to test with compression and encryption?

Reply:
I mean we should delete self.pr('CREATING') It is useless.